nautobot/core/utils/data.py

```diff
@@ -100,7 +100,11 @@ def render_jinja2(template_code, context):
     """
     rendering_engine = engines["jinja"]
     template = rendering_engine.from_string(template_code)
-    return template.render(context=context)
+    # For reasons unknown to me, django-jinja2 `template.render()` implicitly calls `mark_safe()` on the rendered text.
```

Comment:
We probably want to update the docstring for this method to state that this is for templating untrusted template code. The typical use case for jinja template rendering is to trust the template code but not the user input (the context variable here). Since this is not a typical use case we should document it.

Also, if we update the docstring we could get rid of this comment and just run the rendered content through  `django.utils.html.escape`

Reply:
Actually, I prefer your method here in case a developer needs to work with the html in the rendered template. We don't want to prematurely escape the content.